src/Tokenizer/Tokens.php

```diff
@@ -1139,6 +1157,26 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
     }
 
     /**
+     * Finds the line number of the first token previous to the one at the specified index, including itself.
+     *
+     * @param int $index Token index
+     *
+     * @return int Line number
+     */
+    private function findNearestPrevTokenLineNumber($index)
```

Comment:
I have not mixed any concerns here, the code is separated just as it was before. You are correct that there are some limitations of the line tracking code but those limitations suit the current use case, which is to track the lines of the source file.

If there becomes a use case to track the lines of the modified file then this code could be extended to provide that improvement. For now it is not necessary, and I suspect that requirement will never exist, so it is certainly not worth the extra work at this stage.